### PR TITLE
Create situations legenda

### PR DESCRIPTION
**Pull request by VWSCoronaDashboard10**
I had to refactor some parts in order to get the choropleth legenda to be responsive. Also a bug popped up in the `use-choropleth-color-scale` which has been fixed.

image

### CODE REVIEW

**Review by VWSCoronaDashboard1 — APPROVED** [review]

File: packages/app/src/components/choropleth-legenda.tsx [hunk 1]
```diff
@@ -1,36 +1,52 @@
...
+          const isLast = index === thresholds.length - 1;
+          return (
+            <Item key={color + threshold}>
+              <LegendaColor color={color} first={isFirst} last={isLast} />
+              <Label
+                css={isFirst ? css({ transform: 'translateX(0%)' }) : undefined}
+              >
+                {label ?? threshold}
+              </Label>
+              {endLabel && isLast && (
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components/choropleth-legenda.tsx:42]: Why not move this out of the `map` structure?

File: packages/app/src/components/choropleth-legenda.tsx [hunk 2]
```diff
@@ -1,36 +1,52 @@
...
-          </Item>
-        ))}
+        {thresholds.map(({ color, threshold, label }, index) => {
+          const isFirst = index === 0;
+          const isLast = index === thresholds.length - 1;
+          return (
+            <Item key={color + threshold}>
+              <LegendaColor color={color} first={isFirst} last={isLast} />
+              <Label
+                css={isFirst ? css({ transform: 'translateX(0%)' }) : undefined}
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components/choropleth-legenda.tsx:38]: ~~What does this magic `translateX(0%)` do? Might be good to add a comment?~~

I see it's an override of the other transforms. Perhaps pass as an alignment prop to the Label component.

File: packages/app/src/components/choropleth-legenda.tsx [hunk 3]
```diff
@@ -1,36 +1,52 @@
...
+  endLabel?: string;
 }
 
 export function ChoroplethLegenda({
   title,
   thresholds,
   valueAnnotation,
+  endLabel,
 }: ChoroplethLegendaProps) {
+  const { width = 0, ref } = useResizeObserver<HTMLSpanElement>();
```
Comment (VWSCoronaDashboard4) [inline packages/app/src/components/choropleth-legenda.tsx:22]: Wouldn't this be a possible source for odd loops? You are observing the size of a node that is restricted in size a parent node that you control the margins of, causing a change in the size of the node ... etc?

File: packages/app/src/components/choropleth-legenda.tsx (hunk 2 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components/choropleth-legenda.tsx:38]: yep, good idea 👍

File: packages/app/src/components/choropleth-legenda.tsx (hunk 3 above)
Comment (VWSCoronaDashboard10) [reply packages/app/src/components/choropleth-legenda.tsx:20]: Nope, the ref is placed on another element so they're not connected.

**Pull request merged**